API/Repositories/Interfaces/ITournamentsRepository.cs

```diff
@@ -5,40 +5,59 @@ namespace API.Repositories.Interfaces;
 
 public interface ITournamentsRepository : IRepository<Tournament>
 {
-    public Task<IEnumerable<Tournament>> SearchAsync(string name);
-
     /// <summary>
-    ///  Creates or udpates a tournament from a web submission.
+    /// Get a <see cref="Tournament"/> entity
     /// </summary>
-    /// <param name="wrapper">The user input required for this tournament</param>
-    /// <param name="updateExisting">Whether to overwrite values for an existing occurrence of this tournament</param>
-    /// <returns></returns>
-    public Task<Tournament> CreateOrUpdateAsync(
-        TournamentWebSubmissionDTO wrapper,
-        bool updateExisting = false
-    );
+    /// <param name="id">Primary key</param>
+    /// <param name="full">Whether to eagerly load navigational properties</param>
+    Task<Tournament?> GetAsync(int id, bool full = false);
 
+    /// <summary>
+    /// Returns true if an entity with the given name and mode exists
+    /// </summary>
     public Task<bool> ExistsAsync(string name, int mode);
-    public Task<PlayerTournamentTeamSizeCountDTO> GetPlayerTeamSizeStatsAsync(
+
+    /// <summary>
+    /// Search for a tournament by name
+    /// </summary>
+    public Task<IEnumerable<Tournament>> SearchAsync(string name);
+
+    /// <summary>
+    /// Create team size statistics for a player
+    /// </summary>
+    /// <param name="playerId">Id (primary key) of target player</param>
```

Comment:
primary key is implied

Reply:
Did this initially but changed my mind, guess I didn't catch all the instances